### DISCUSSION TITLE
Client-side exception error

### DISCUSSION THREAD

**Discussion by Vodolay** [vercel/next.js#58799] [Help]
### Summary

In my typescript project a client page importing the package jstat was working perfectly fine until v.14.0.1
As of v.14.0.2 a client page gives the following error:

"Application error: a client-side exception has occurred (see the browser console for more information)."

And the browser console says:
"Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'random') .... "

How do I switch to 14.0.2 or higher and defeat this error?

### Additional information

```js
The jstat package doesn't have types, but I have exempted it from type checking in declarations.d.ts file.
```

**Comment by icyJoseph** [comment]
This TypeError doesn't mean a TypeScript error, it means that, it tried to do something like `Math.random`, for example, but somehow `Math` was undefined, at runtime...

Which is really hard to believe, or rather, understand... could you share a screenshot of the error stack?

**Comment by Vodolay** [comment]
The error only occurs at deployment after pnpm run start (pnpm run build runs with no complaints). Under pnpm run dev there is no error either. Here are snapshots of relevant parts of the error stack from the dev console:

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'random')
    at 976-ea927a2fdbd431b9.js:1:12397
    at 9497 (976-ea927a2fdbd431b9.js:1:17723)
    at s (webpack-dff8a11e62dcc0fc.js:1:152)
    at 6744 (page-9bfce8e4dcbb5a43.js:1:230)
    at Function.s (webpack-dff8a11e62dcc0fc.js:1:152)
    
  The screenshots of the relevant parts of the files mentioned above are attached.
Firs line:
![1](https://github.com/vercel/next.js/assets/37164020/62870096-1e18-4eb3-b02f-d20251f611dc)
Second line:
![2](https://github.com/vercel/next.js/assets/37164020/a5130063-f1aa-4a9c-8033-f86fdc1c56f2)
Third line:
![3](https://github.com/vercel/next.js/assets/37164020/628c1aad-f842-47ae-a4f3-d767b73f3c4c)
Fourth line:
![4](https://github.com/vercel/next.js/assets/37164020/929c75bb-d5a1-447e-b038-1a8a8cb37978)
Fith line:
![5](https://github.com/vercel/next.js/assets/37164020/500ac3dc-18f1-406c-968c-829681b4718a)

And there is no error with next.js v.14.0.1. However versions 14.0.2 and 14.0.3 produce this error.

**Reply by icyJoseph** [reply]
🤔 could you try to set up a small reproduction repository? If the error happens between those releases we could try to figure out which canary broke it.

I would also try to disable swcMinify, and see if that helps, just in case.

**Reply by Vodolay** [reply]
In this stripped-down repo the error is reproducible:

https://github.com/Vodolay/canary

**Reply by icyJoseph** [reply]
@Vodolay that was not a minimum reproduction repository, but I've got to see the error on another project. As expected, setting `swcMinify` to false, does fix the issue. 

So this has been introduced when Next.js upgraded their SWC compiler. Now the question is... how do we make a reproducible code example in the SWC playground....

**Comment by Vodolay** [comment]
@icyJoseph thank you, for pointing out where the issue is.

I do believe that the repo is functional: please go to the front page and then click on the only phrase in the body and you will see the error message. The code which generates it is inside one of the two imported modules.

**Reply by icyJoseph** [reply]
Oh yeah, for sure, but it is not minimal, you know, it has a bunch of other stuff, it should be only and exactly what breaks.

But it doesn't matter. I have independently verified it. The problem is that I don't have time to figure out which SWC settings are responsible, but if we had a repository, we could at least share that with the SWC team and they could try to figure it out.
